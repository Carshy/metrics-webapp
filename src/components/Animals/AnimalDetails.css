@media screen and (max-width: 768px) {
  .animal-info {
    position: fixed;
    left: 10%;
    background-color: #4369b2;
    overflow-y: scroll;
    height: 90vh;
    width: 60%;
    border-radius: 30px;
    padding-top: 20px;
    color: #fff;
    opacity: 0.7;
  }
}

.animal-info {
  position: fixed;
  right: 300px;
  overflow-y: scroll;
  height: 90vh;
  width: 60%;
  padding-top: 20px;
  color: #fff;
  border: 1px solid rgba(255, 255, 255, 0.3);
  background: rgba(255, 255, 255, 0.2);
  border-radius: 18px;
  box-shadow: 0 4px 30px rgba(0,0,0,0.1);
  backdrop-filter: blur(5px);
}

.animal-title {
  font-size: 34px;
  text-align: center;
}

.animal-image {
  display: flex;
  justify-content: center;
  align-items: center;
  margin: 2% 0 2% 0;
}

.animal-image img {
  width: 54rem;
  height: 24rem;
  border-radius: 8px;
}

.animal-characteristics {
  font-size: 28px;
  text-align: center;
  margin-bottom: 2%;
}

.span-details {
  margin: 0 0 0 15%;
  font-size: 18px;
}

.spanned-right {
  margin-left: 51%;
}

.spanned {
  margin-bottom: 1.2rem;
}
