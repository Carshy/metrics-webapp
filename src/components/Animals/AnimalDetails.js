import React, { useState, useEffect } from 'react';
import { useSelector } from 'react-redux';
import { useParams } from 'react-router-dom';
import './AnimalDetails.css';

const AnimalDetails = () => {
  const { id } = useParams();
  const animals = useSelector((state) => state.animals);
  const [animal, setAnimal] = useState({});

  useEffect(() => {
    animals.forEach((animal) => {
      if (parseInt(animal.id, 10) === parseInt(id, 10)) {
        setAnimal(animal);
      }
    });
  }, [id, animals]);

  return (
    <div className="animal-details-section">
      <div className="animal-info">
        <h2 className="animal-title">{animal.name}</h2>
        <div className="animal-image">
          <img src={animal.image_link} alt={`${animal.name} details`} />
        </div>
        <h2 className="animal-characteristics">Characteristics</h2>
        <div className="span-details">
          <p className="spanned">
            Latin Name
            <span className="spanned-right">{animal.latin_name}</span>
          </p>
          <p className="spanned dark">
            Animal Type
            <span className="spanned-right">{animal.animal_type}</span>
          </p>
          <p className="spanned">
            Active Times
            <span className="spanned-right">{animal.active_time}</span>
          </p>
          <p className="spanned dark">
            Minimum Length
            <span className="spanned-right">{animal.length_min}</span>
          </p>
          <p className="spanned">
            Maximum Length
            <span className="spanned-right">{animal.length_max}</span>
          </p>
          <p className="spanned dark">
            Minimum Weight
            <span className="spanned-right">{animal.weight_min}</span>
          </p>
          <p className="spanned">
            Maximum Weight
            <span className="spanned-right">{animal.weight_max}</span>
          </p>
          <p className="spanned dark">
            Life span
            <span className="spanned-right">{animal.lifespan}</span>
          </p>
          <p className="spanned">
            Habitats
            <span className="spanned-right">{animal.habitat}</span>
          </p>
          <p className="spanned dark">
            Diet
            <span className="spanned-right">{animal.diet}</span>
          </p>
          <p className="spanned">
            Locations
            <span className="spanned-right">{animal.geo_range}</span>
          </p>
        </div>
      </div>
    </div>
  );
};

export default AnimalDetails;
