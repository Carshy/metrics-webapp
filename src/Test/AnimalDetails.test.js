import React from 'react';
import Renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import AnimalDetails from '../components/Animals/Animal';

test('renders correctly', () => {
  const tree = Renderer.create(
    <React.StrictMode>
      <Provider store={store}>
        <BrowserRouter>
          <AnimalDetails />
        </BrowserRouter>
      </Provider>
    </React.StrictMode>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
