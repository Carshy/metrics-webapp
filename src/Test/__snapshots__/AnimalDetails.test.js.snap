// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container-section"
>
  <div
    className="animalia-info-div"
  >
    <div
      className="animalia-dropdown"
    >
      <img
        className="animal-images"
      />
      <a
        href="/details/undefined"
        onClick={[Function]}
      >
        <img
          alt="arrow"
          className="arrow-round"
          src="arrow1.PNG"
        />
      </a>
    </div>
    <div
      className="animalia-description"
    >
      <h2 />
      <h3 />
    </div>
  </div>
</div>
`;
